### PR TITLE
Feature/add models question and answer

### PR DESCRIPTION
**Pull request by bogdanrymar**
Create models Question and Answer. Add validations. Create tests for them. Create factories and seeds

### CODE REVIEW

File: app/models/question.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+
+class Question < ApplicationRecord
+  has_many :answers, dependent: :destroy
+
+  validates :title, presence: true, length: { minimum: 2, maximum: 255 }
+  validates :body, presence: true, length: { minimum: 2, maximum: 1000 }
+  validates :user_name, presence: true, length: { minimum: 2, maximum: 255 }
+  validates :user_email, length: { maximum: 255 }
```
Comment (viktoriia-horobets) [inline app/models/question.rb:9]: validate against regex

File: db/seeds.rb [hunk 2]
```diff
@@ -39,6 +39,22 @@
   )
 end
 
+25.times do
+  Question.create!(
+    title:  Faker::Lorem.sentence ,
+    user_name:  Faker::Internet.user_name ,
+    user_email:  Faker::Internet.email ,
+    body:  Faker::Lorem.paragraph
+  )
```
Comment (SvitDZHUS) [inline db/seeds.rb:48]: Reuse your factory here not to specify all attributes one more time.

File: db/seeds.rb [hunk 3]
```diff
@@ -39,6 +39,22 @@
...
+end
+
+25.times do
+  Answer.create!(
+    question_id: Question.all.sample.id,
+    body:  Faker::Lorem.paragraph
+  )
```
Comment (SvitDZHUS) [inline db/seeds.rb:55]: Use factory

File: app/models/question.rb (hunk 1 above)
Comment (bogdanrymar) [reply app/models/question.rb:9]: I created validation in the new commit

File: db/seeds.rb (hunk 3 above)
Comment (bogdanrymar) [reply db/seeds.rb]: done

File: db/migrate/20220724120525_create_questions.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+class CreateQuestions < ActiveRecord::Migration[6.0]
+  def change
+    create_table :questions do |t|
+      t.string :title
+      t.string :user_name
+      t.string :user_email
+      t.text :body
+
+      t.timestamps
```
[the remark this answers is not in the document]
Comment (bogdanrymar) [reply db/migrate/20220724120525_create_questions.rb:9]: I added validation in the new commit

File: spec/models/question_spec.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  describe 'validations' do
+    it { is_expected.to validate_presence_of(:title) }
+    it { is_expected.to validate_length_of(:title).is_at_least(2).is_at_most(255) }
+    it { is_expected.to validate_presence_of(:body) }
+    it { is_expected.to validate_length_of(:body).is_at_least(2).is_at_most(1000) }
+    it { is_expected.to validate_presence_of(:user_name) }
+    it { is_expected.to validate_length_of(:user_name).is_at_least(2).is_at_most(255) }
+    it { is_expected.to validate_length_of(:user_email).is_at_most(255) }
+    let(:question_invalid_email) { Question.new(user_name: 'abc', user_email: '123.com', body: 'abc', title: 'abc') }
+    it 'can be saved' do
```
Comment (SvitDZHUS) [inline spec/models/question_spec.rb:15]: the test description does not match the test logic

File: spec/models/question_spec.rb [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+RSpec.describe Question, type: :model do
+  describe 'validations' do
+    it { is_expected.to validate_presence_of(:title) }
+    it { is_expected.to validate_length_of(:title).is_at_least(2).is_at_most(255) }
+    it { is_expected.to validate_presence_of(:body) }
+    it { is_expected.to validate_length_of(:body).is_at_least(2).is_at_most(1000) }
+    it { is_expected.to validate_presence_of(:user_name) }
+    it { is_expected.to validate_length_of(:user_name).is_at_least(2).is_at_most(255) }
+    it { is_expected.to validate_length_of(:user_email).is_at_most(255) }
+    let(:question_invalid_email) { Question.new(user_name: 'abc', user_email: '123.com', body: 'abc', title: 'abc') }
```
Comment (SvitDZHUS) [inline spec/models/question_spec.rb:14]: add some spaces

File: spec/models/question_spec.rb (hunk 5 above)
Comment (bogdanrymar) [reply spec/models/question_spec.rb]: changed

File: app/models/question.rb [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+  validates :user_email, format: { with: URI::MailTo::EMAIL_REGEXP }
```
Comment (mmrshk) [inline app/models/question.rb:10]: why you need two lines for this? simply write it in one line!

File: db/seeds.rb [hunk 8]
```diff
@@ -8,7 +8,7 @@
 #
 #   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
 #   Character.create(name: 'Luke', movie: movies.first)
-
+require 'factory_bot_rails'
```
Comment (mmrshk) [inline db/seeds.rb:11]: 1. remove comments from 3 till 10th line
2. add empty line after 11

File: spec/models/question_spec.rb (hunk 6 above)
Comment (mmrshk) [reply spec/models/question_spec.rb:15]: use factory here

File: spec/models/question_spec.rb [hunk 9]
```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'factory_bot_rails'
```
Comment (mmrshk) [inline spec/models/question_spec.rb:4]: add empty line

Comment (mmrshk) [reply spec/models/question_spec.rb:4]: remove factory_bot_rails here, i think it's redundant

File: spec/models/question_spec.rb [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+  describe 'validations' do
+    it { is_expected.to validate_presence_of(:title) }
+    it { is_expected.to validate_length_of(:title).is_at_least(2).is_at_most(255) }
+    it { is_expected.to validate_presence_of(:body) }
+    it { is_expected.to validate_length_of(:body).is_at_least(2).is_at_most(1000) }
+    it { is_expected.to validate_presence_of(:user_name) }
+    it { is_expected.to validate_length_of(:user_name).is_at_least(2).is_at_most(255) }
+    it { is_expected.to validate_length_of(:user_email).is_at_most(255) }
+
+    let(:question_with_invalid_email) { FactoryBot.build(:question, :question_with_invalid_email) }
```
Comment (mmrshk) [inline spec/models/question_spec.rb:15]: update to `{ build(:question, :question_with_invalid_email) }`, don't use FactoryBot

**Review by mmrshk — APPROVED** [review]
lgtm

**Pull request merged**